core/src/err.rs

```diff
@@ -9,8 +9,8 @@ pub type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Error, Debug, Clone)]
 pub enum Error {
-    #[error(transparent)]
-    BsonDeserialization(#[from] bson::de::Error),
+    #[error("Parsing metadata for {0} gave: {1}")]
```

Comment:
[nit]
I guess we are only deserializing metadata type information, so it makes sense. But since the error message is user facing, I don't know if they need that level of details.
What about "Retrieving information" instead of "Parsing metadata"?